docs/internals.md

```diff
@@ -65,13 +65,75 @@ retries.
 
-## FlushSessions
+### FlushSessions
 
 The FlushSessions state (class `BeaconSendingFlushSessionsState`) is used to send all
 data which has not been transferred so far to the server.
 
-## Terminal
+### Terminal
 
 The Terminal state (class `BeaconSendingTerminalState`) is the last state in OpenKit's internal 
 state machine. After this state is reached the background thread responsible for sending data 
 is terminated gracefully.
+
+## Data Capturing (BeaconCache)
+
+To be able to capture events while OpenKit is still initializing, all reported events are stored
+in a cache, the so called BeaconCache.
+
+An illustration how this cache looks like can be found below, followed by a detailed explanation.
+
+![diagram](./pics/OpenKit-BeaconCache.svg)
+
+### Basic BeaconCache Layout
+
+The BeaconCache is a 2-dimensional cache, storing data for each Session, internally also referred to
+as Beacon. Each OpenKit instance has such a cache, so if your application starts multiple OpenKit instances
+in parallel, keep in mind that each OpenKit instance consumes cache memory.  
+
+The example image shows three Sessions started from one OpenKit instance and for each Session a different number of Event Data
+and Action Data. The distinction between Event Data and Action Data is only used to determine the sending
+order of captured events.
+
+### BeaconCache Records
+
+A record is a single captured event, like an Action, a Web Request or anything else captured with
+OpenKit. A record is already serialized data which can be sent to the backend system.
+
+### BeaconCache Eviction
+
+By default the BeaconCache has two Eviction strategies, which are triggered whenever new data
+is inserted. Triggering such a strategy does not necessarily mean that records are evicted from the cache, but rather
+the strategy evaluates whether it makes sense to run or not.
+
+The eviction strategies are ran in a separate background thread, which is started when OpenKit is started and
+shut down when OpenKit is terminated.
+
+#### Time Based Eviction
+
+Since Dynatrace/AppMon backend services will throw away too old data, it make sense to not send such data to the
+backend system.
+By default records that are older than 1 hour and 45 minutes are thrown away, but when initializing an
+OpenKit instance via the builder the value can be set by calling `withBeaconCacheMaxRecordAge` with a long argument specifying the
+maximum record age in milliseconds. Bear in mind that Dynatrace/AppMon might still throw away the data, so setting
+this value too high might have no effect.
+
+It is possible to disable this strategy by setting the argument to `withBeaconCacheMaxRecordAge` to a value less than
+or equal to 0.
+
+#### Size based Eviction
+
+The second eviction strategy is used to keep the memory consumed by an OpenKit instance within certain boundaries.
+This strategy starts to run when the memory consumed by the BeaconCache exceeds a certain upper threshold and stops
```

Comment:
This strategy starts to run ... -> This strategy evicts beacons if the memory usage of the BeaconCache exceeds a configured upper bound and